packages/@aws-cdk/integ-tests/lib/assertions/deploy-assert.ts

```diff
@@ -42,54 +34,83 @@ export class DeployAssert extends CoreConstruct {
    * Finds a DeployAssert construct in the given scope
    */
   public static of(construct: IConstruct): DeployAssert {
-    const scopes = Node.of(construct).scopes.reverse();
+    const scopes = Node.of(Node.of(construct).root).findAll();
     const deployAssert = scopes.find(s => DeployAssert.isDeployAssert(s));
     if (!deployAssert) {
       throw new Error('No DeployAssert construct found in scopes');
     }
     return deployAssert as DeployAssert;
   }
 
-  /** @internal */
-  public readonly _assertions: IAssertion[];
-
   constructor(scope: Construct) {
-    super(scope, 'DeployAssert');
+    scope = new Stack(scope, 'DeployAssert');
+    super(scope, 'Default');
```

Comment:
This is weird to me. Why is the Stack not inside the `DeployAssert` ?

We can hold on to the stack in a `private readonly scope` so that we can still create other resources inside the right Stack. But the swapparoo here is a little confusing.

Reply:
It's so that we don't have to create `DeployAssert` in the scope of a stack. I was originally going to make `DeployAssert` extend `Stack` but I thought it would be nice if `DeployAssert` could be treated like a stack, but the user wouldn't be able to access any of the stack functionality (the methods they would see would just be the `DeployAssert methods and not any `Stack` methods).

So you could create any of the assertion resources by passing in `DeployAssert` as the scope

```ts
const deployAssert = new DeployAssert(app);
new AwsApiCall(deployAssert, 'AwsApiCall', {...});
```

We could change it to what you suggested and make it a `public readonly stack` which the user would have to know to use


```ts
const deployAssert = new DeployAssert(app);
new AwsApiCall(deployAssert.stack, 'AwsApiCall', {...});
```